### DISCUSSION TITLE
Adding Model Flag when generating filament resources

### DISCUSSION THREAD

**Discussion by ousid** [filamentphp/filament#5881] [Ideas]
I have been using the `--generate` flag for a quite a while with `make:filament-resource` command, and it's been really useful, but I noticed something while working with the command.

## The Main Obstacle
Sometimes you need to structure the resources under groups, but the `Model` namespace is not like the same group as the resource, for example we have the following structure

```
📦Resources
 ┗ 📂Users
 ┃ ┣ 📂RoleResource
 ┃ ┃ ┗ 📂Pages
 ┃ ┃ ┃ ┣ 📜CreateRole.php
 ┃ ┃ ┃ ┣ 📜EditRole.php
 ┃ ┃ ┃ ┗ 📜ListRoles.php
 ┃ ┗ 📜RoleResource.php
```

But the `Role` Model, is in another namespace (e.g. `Spatie\LaravelPermissions\Model\Role`)

If you want to do the following:
```bash
php artisan make:filament-resource --generate
```
Then you pass the namespace (with group) 
```bash
Users/Role
```

The command will search for a `Model` under the same namespace, to generate the __columns__, from. Which does not exist.

## The Potential Solution
I believe it's better to add `--model` flag to set the model namespace path, if the `ModelResource` and the `Model` does not share the same namespace path.

## The Benefits of This Approach
- It's better to organize, and separate the logic, of the `ModelResource` and the `Model` itself
- If the both has different namespaces, it will be a lot easier to control, with less time organizing, and modifying the generated resource.

I think this approach will enhance the __DX__, and making the development workflow more productive.

Let me know what do you think, and if it's a good idea, I'm happy to submit a `PR` for it!

__PS__: I switched the logic in here, the command will give you an input for the `Model` namespace path, and it generates the resource upon the path, the idea in here is to generate the resource path along with the `Model`.

cc: @danharrin

**Comment by danharrin** [comment]
I'm a fan of nested models within namespaces, whats the difference?

**Reply by ousid** [reply]
Well, the difference, that maybe you are not nesting your models, in subdirectories, and just all of your models are in the same level, in the other hand, you want to categorize your filament resources under subdirectories.

In fact, it's not that a big issue, you can move files, and do some [find and replace] for namespaces, but I thought it can enhance the DX in this case.

What do you think?

**Reply by danharrin** [reply]
I think having a --model flag is fine, but personally I don't see why you would nest resources but not nest the models themselves?

**Reply by ousid** [reply]
Fair Point.
Sometimes, you may add filament into a legacy/stand codebase, and the model already structured, in a certain way, by the developers who worked on the codebase. So, it's hard to re-structure the models, and edit the namespace all over the app, it will be time-consuming...

**Comment by CarlosDuarte17** [comment]
I have a project where the models are not in the default app/Models directory, so this approach is useful.

**Reply by danharrin** [reply]
I'm happy for a --model flag to be added through a PR :)
